sys/src/9/port/devufs.c

```diff
@@ -139,22 +139,41 @@ ufsread(Chan *c, void *a, int32_t n, int64_t offset)
 }
 
 static void
-mount(const char* path)
+mount(char* a, int32_t n)
 {
-	if (!path || strlen(path) == 0) {
-		error(Enonexist);
+	Proc *up = externup();
+
+	Cmdbuf* cb = parsecmd(a, n);
+	if (waserror()) {
+		free(cb);
+		nexterror();
+		print("couldn't parse mount path: %s: %r", a);
+		return;
+	}
+	poperror();
+
+	// TODO Make namec and related functions accept const char*
+	Chan* c = namec(cb->buf, Aopen, ORDWR, 0);
+	free(cb);
+	cb = nil;
+	if (waserror()) {
+		cclose(c);
+		nexterror();
+		print("couldn't open file %s to mount: %r", cb->buf);
 		return;
 	}
+	poperror();
 
-	print("Mount %s\n", path);
-	struct mount* mp = mallocz(sizeof(struct mount), 1);
+	struct mount* mp = newufsmount(c);
 	int rcode = ffs_mount(mp);
```

Comment:
could you have ffs_mount call error? this would really simplify the logic here. This is kind of a mix of error()-style and Unix-style logic. It would be nice if you could use error() in the plan 9 style. This would mean changing ffs_mount to call error()